src/routers/UserRouter.ts

```diff
@@ -362,6 +362,92 @@ const getAdminRouter = (basePath: string = '/', routePath: string = '/management
     })
   );
 
+  router.put(
+    path,
+    guard.enforcePrimaryGroup(true),
+    AuthzGuards.writeUsersGuard,
+    asyncHandler(async (req: AuthzRequest, res: Response) => {
+      const authzService: AuthzServiceSpec = req.app.locals.authzService;
+      const authMgmtService: AuthManagementService = req.app.locals.authManagementService;
+
+      const emails = get(req.body, 'emails', []);
+      const groups = get(req.body, 'groups', []);
+
+      const result: {
+        [email: string]: {
+          success: boolean;
+          error?: string;
+        };
+      } = {};
+
```

Comment:
can we do a "fail-fast" to validate all emails before we continue with the processing
```
      const usersIds = await forEachAsync(emails, async (email: string) => {
        validateEmail(email);
        const user = await authMgmtService.getUserByEmail(email);
        const userId = get(user, 'user_id');

        if (req.identity.sub === userId) {
          throw new UnauthorizedError('You cannot update your own user.', 403);
        }
        return userId
      });
```